tests/test_three_d.py

```diff
@@ -16,6 +16,30 @@
 
 logging.set_log_level(logging.LogLevel.SILENT)
 
+class TestSharedSurface(unittest.TestCase):
+    def test_surface_with_shared_elements(self):
```

Comment:
I don't get the naming of these tests: why 'shared surface' and 'shared elements' ?